applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -811,18 +833,18 @@ static void discover_cb(struct bt_conn *conn, int err, enum bt_audio_dir dir)
 	}
 
 	for (int i = 0; i < CONFIG_CODEC_CAP_COUNT_MAX; i++) {
-		if (dir == BT_AUDIO_DIR_SINK) {
+		if (dir == BT_AUDIO_DIR_SINK && !err) {
```

Comment:
No need to check for `&& !err` here or below, as this is done on line 830.

Reply:
ret and err is not the same here. Err is an argument coming into the callback from host